collectors/30/kafka_offset_collector.py

```diff
@@ -66,26 +92,34 @@ def get_partitions(topic):
 def report():
     # see the monitoring.json chef role
     zk_quorums = os.getenv('MONITORED_ZOOKEEPER_QUORUMS')
-    kafka_brokers = os.getenv('KAFKA_BROKERS')
 
     if zk_quorums is None:
         raise RuntimeError('MONITORED_ZOOKEEPER_QUORUMS not found')
 
-    if kafka_brokers is None:
-        raise RuntimeError('KAFKA_BROKERS not found')
-
-    # we could just use ZK to find all topics, should we just do that?
-    topics = ['mobile_metrics', 'raw_events']
-
     consumer_group = 'secor_group'
 
     for zk_quorum in zk_quorums.split('|'):
         zk = KazooClient(hosts=zk_quorum)
-        kafka = KafkaClient(kafka_brokers)
-
         zk.start()
+
+        kafka_brokers = get_kafka_brokers(zk)
+        if not kafka_brokers:
+            raise RuntimeError('KAFKA_BROKERS could not be fetched from ZK')
+        kafka_init = 'env=production kafka=%s zk=%s' % (','.join(kafka_brokers), zk_quorum)
+        kafka = KafkaClient(kafka_init)
+
+        # Pull topics from zookeeper
+        # e.g. topics = ['mobile_metrics', 'raw_events']
+        topics = get_kafka_topics(zk)
+        if not topics:
+            raise RuntimeError('KAFKA_TOPICS could not be fetched from ZK')
         for topic in topics:
-            total_broker_offset = report_broker_info(kafka, zk, topic)
+            try:
+                total_broker_offset = report_broker_info(kafka, zk, topic)
+            except LeaderNotAvailableError as err:
+                logging.exception(err)
+                total_broker_offset = 0
+
```

Comment:
Both Kazoo and Kafka just shut down the connection and close the socket which will happen on process exit anyway.